expression/builtin_time.go

```diff
@@ -2505,18 +2505,59 @@ type timeFormatFunctionClass struct {
 }
 
 func (c *timeFormatFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinTimeFormatSig{newBaseBuiltinFunc(args, ctx)}
-	return sig.setSelf(sig), errors.Trace(c.verifyArgs(args))
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpDuration, tpString, tpString)
```

Comment:
I messed them up. According to your [comment](https://github.com/pingcap/tidb/pull/4051#discussion_r132370983) It should 
`bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpString, tpDuration, tpString)`